backend/src/positions/neck_position.py

```diff
@@ -129,14 +125,10 @@ def shift(self, shift: int, max_finger: int = 4) -> None:
                 f"Cannot shift position by {shift} fingers, "
                 f"min finger {min(self.fingers)} would be exceeded."
             )
-        new_finger_positions: list[FingerPosition] = []
-        for finger_position in self._finger_positions:
-            if finger_position.finger > 0:
-                new_finger_positions.append(
-                    FingerPosition(finger_position.placement, finger_position.finger + shift)
-                )
-            else:
-                new_finger_positions.append(finger_position)
+        new_finger_positions: list[FingerPosition] = [
+            FingerPosition(fp.placement, fp.finger + shift) if fp.finger > 0 else fp
+            for fp in self._finger_positions
+        ]
         self._finger_positions = new_finger_positions
```

Comment:
Not very important as it is still clear, but maybe don't use a temporary variable here as it does not do anything
```suggestion
        self._finger_positions = [
            FingerPosition(fp.placement, fp.finger + shift) if fp.finger > 0 else fp
            for fp in self._finger_positions
        ]
```